io/src/main/scala/sbt/internal/io/SourceModificationWatch.scala

```diff
@@ -97,12 +97,12 @@ private[sbt] final class WatchState private (
     val newKeys =
       fs.filter(Files.exists(_)).foldLeft(registered) {
         case (ks, d) if Files.isDirectory(d) =>
-          if (registered.contains(d)) ks
+          if (ks.contains(d)) ks
```

Comment:
Is this considered a bug fix or an improvement?
If so, is there an epic issue that you can link to, etc?

Reply:
I'm not 100% certain it would fix a specific buggy use case, but it's an improvement on the status quo: which is that sometimes you'll double register the same directory, which probably leads to slower performance.

Just from a code standpoint would you not think it's wrong to be using the initial ("zero") value of the `foldLeft` rather than the accumulator?

If you're concerned about this causing regression it can't because it's only _not_ double registering files, it doesn't cause some files to no longer be registered.

Reply:
The change itself LGTM, but I'd like you to change the commit message to reflect the nature of this fix, mostly so I can roll it up into a release note later.

Reply:
sure, will do